docs/changelog/124865.yaml

```diff
@@ -0,0 +1,7 @@
+pr: 124865
+summary: "Permanently switch from Java SecurityManager to Entitlements.
+  With JDK24, the Java SecurityManager has been completely disabled. In order to retain an additional protection
+  mechanism, Elasticsearch is replacing SecurityManager with our own protection mechanism, Entitlements."
```

Comment:
optional nit
```suggestion
  With JDK24, the Java SecurityManager has been completely disabled. In order to retain a similar protection
  mechanism, Elasticsearch is replacing SecurityManager with its own protection mechanism, Entitlements."
```

Reply:
Wondering, this makes it sound a bit like entitlements are just a JDK 24 thing. Should we explicitly mention that this is the case for all versions of ES supporting JDK 24 regardless of which JDK version used at runtime?

Reply:
Maybe like this?
```suggestion
  With JDK24, the Java SecurityManager has been completely disabled. In order to retain a similar protection
  mechanism, Elasticsearch is replacing SecurityManager with its own protection mechanism, Entitlements. 
  This is the case for all versions of Elasticsearch supporting JDK24 regardless of the JDK version used at runtime."
```

Reply:
I rephrased it so the focus in not on JDK 24 anymore, and stating we are making the switch permanently. But let me know if you think I should still add "This is the case for all versions of Elasticsearch supporting JDK24 regardless of the JDK version used at runtime."

Reply:
that's great 👍